Xamarin.Forms.Platform.WinRT/Platform.cs

```diff
@@ -761,8 +761,35 @@ void OnPageActionSheet(Page sender, ActionSheetArguments options)
 				dialog.CancelCommandIndex = (uint)dialog.Commands.Count - 1;
 			}
 
-			IUICommand command = await dialog.ShowAsync();
+			IUICommand command = await dialog.ShowAsyncQueue();
 			options.SetResult(command.Label == options.Accept);
 		}
 	}
+	
+	// refer to http://stackoverflow.com/questions/29209954/multiple-messagedialog-app-crash for why this is used
+	// in order to allow for multiple MessageDialogs, or a crash occurs otherwise
+	public static class MessageDialogExtensions
+	{
+		static TaskCompletionSource<MessageDialog> _currentDialogShowRequest;
+
+		public static async Task<IUICommand> ShowAsyncQueue(this MessageDialog dialog)
+		{
+			if (!Window.Current.Dispatcher.HasThreadAccess)
```

Comment:
`Window.Current` isn't a good way to make this check here - if the user tries to call `DisplayAlert()` from a non-UI thread, `Window.Current` could be `null` and they'll get a `NullReferenceException`. 

Instead of `if (!Window.Current.Dispatcher.HasThreadAccess)` you could check `if (Device.IsInvokeRequired)`, which doesn't have that issue.